### PR TITLE
dua-cli: update to 2.16.0

### PR DESCRIPTION
**Pull request by cinerea0**
#### Testing the changes
- I tested the changes in this PR: **YES**

---

Closes #33954

### CODE REVIEW

File: srcpkgs/dua-cli/patches/0001-PATCH-fix-on-platforms-without-64-bit-atomics.patch [hunk 1]
```diff
@@ -0,0 +1,31 @@
+From e49a4bfc495e347834f7b3e8fc3a1501a0b313d7 Mon Sep 17 00:00:00 2001
```
Comment (ericonr) [inline srcpkgs/dua-cli/patches/0001-PATCH-fix-on-platforms-without-64-bit-atomics.patch:1]: Could you rename the file and remove the header differences to minimize the patch?

If you think they could take this upstream too, that'd be great!

**Comment by cinerea0** [comment]
PR to potentially get the patch merged upstream: https://github.com/Byron/dua-cli/pull/117

**Comment by cinerea0** [comment]
My upstream PR has been merged and a new version has been released; this PR has been updated to reflect that.

**Comment by ericonr** [comment]
Nice!

**Pull request merged**